protocol/xmrtaker/recovery.go

```diff
@@ -47,18 +47,19 @@ func NewRecoveryState(b backend.Backend, dataDir string, secret *mcrypto.Private
 
 	ctx, cancel := context.WithCancel(b.Ctx())
 	s := &swapState{
-		ctx:            ctx,
-		cancel:         cancel,
-		Backend:        b,
-		sender:         sender,
-		privkeys:       kp,
-		pubkeys:        pubkp,
-		dleqProof:      dleq.NewProofWithSecret(sc),
-		contractSwapID: contractSwapID,
-		contractSwap:   contractSwap,
-		infoFile:       pcommon.GetSwapRecoveryFilepath(dataDir),
-		claimedCh:      make(chan struct{}),
-		info:           pswap.NewEmptyInfo(),
+		ctx:               ctx,
+		cancel:            cancel,
+		Backend:           b,
+		sender:            sender,
+		privkeys:          kp,
+		pubkeys:           pubkp,
+		dleqProof:         dleq.NewProofWithSecret(sc),
+		moneroBlockHeight: 0, // TODO: Can we optimise this?
```

Comment:
This was the only line that changed.  If we get the block height before the swap into the recovery information, we could speed recovery wallet synchronization.

Reply:
can add this in #203